File: app/src/main/java/org/anvei/novelreader/novel/WebsiteIdentifier.java
package org.anvei.novelreader.novel;

import androidx.annotation.Nullable;

import java.lang.reflect.Field;

public enum WebsiteIdentifier {
    SFACG,
    BIQUMU,
    //YUNXI,
    W147XS;

    public static @Nullable WebsiteIdentifier getIdentifier(String str) {
        str = str.toUpperCase();
        WebsiteIdentifier res = null;
        Class<?> clazz = WebsiteIdentifier.class;
        Field[] fields = clazz.getFields();
        for (Field field : fields) {
            if (str.equals(field.getName())) {
                try {
                    res = (WebsiteIdentifier) field.get(null);
                    break;
                } catch (IllegalAccessException e) {
                    e.printStackTrace();
                }
            }
        }
        return res;
    }
}

File: app/src/main/java/org/anvei/novelreader/novel/website/WebsiteIdentifier.kt
package org.anvei.novelreader.novel.website

enum class WebsiteIdentifier {
    UNKNOWN,  // 未知网站标识符
    SFACG,
    BIQUMU,
    W147XS;

    companion object {
        /**
         * 根据网站表示符的字符形式，获取一个WebsiteIdentifier变量
         */
        @JvmStatic
        fun getIdentifier(str: String): WebsiteIdentifier? {
            val str1 = str.uppercase()
            var res: WebsiteIdentifier? = null
            val clazz: Class<*> = WebsiteIdentifier::class.java
            val fields = clazz.fields
            for (field in fields) {
                if (str1 == field.name) {
                    try {
                        res = field[null] as WebsiteIdentifier
                        break
                    } catch (e: IllegalAccessException) {
                        e.printStackTrace()
                    }
                }
            }
            return res
        }
    }
}